src/BrandwatchReactAuth.js

```diff
@@ -0,0 +1,61 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import jwtDecode from 'jwt-decode';
+import TokenStore from 'viziaauth';
+
+export default class BrandwatchReactAuth extends Component {
+  getChildContext() {
+    return {
+      brandwatchAuthLogout: this.handleLogout,
+    };
+  }
+
+  constructor(props) {
+    super(props);
+    this.store = new TokenStore(this.props.domain),
+    this.handleLogout = this.handleLogout.bind(this);
+
+    this.state = {
+      loggedIn: undefined,
+    };
+
+    props.onCreateStore(this.store);
+  }
+
+  componentWillMount() {
+    this.store.getToken({ aud: this.props.audience }).then((token) => {
+      if (!token) {
+        return window.location.href = this.store.loginUrl;
```

Comment:
> The Location.replace() method replaces the current resource with the one at the provided URL. The difference from the assign() method is that after using replace() the current page will not be saved in session History, meaning the user won't be able to use the back button to navigate to it. - [mdn]

Yeah I think you're right andy, `location.replace` seems like a better solution, though in _theory_ it can throw, so we might cover that

[mdn]: https://developer.mozilla.org/en-US/docs/Web/API/Location/replace